soc/nordic/nrf54h/power.c

```diff
@@ -126,20 +125,32 @@ static void do_suspend_to_ram(void)
 
 	sys_resume();
 }
-#endif /* IS_ENABLED(CONFIG_PM_S2RAM) */
+#endif /* defined(CONFIG_PM_S2RAM) */
 
 void pm_state_set(enum pm_state state, uint8_t substate_id)
 {
 	if (state != PM_STATE_SUSPEND_TO_RAM) {
 		k_cpu_idle();
 		return;
 	}
-#if IS_ENABLED(CONFIG_PM_S2RAM)
+#if defined(CONFIG_PM_S2RAM)
 	do_suspend_to_ram();
 #endif
 }
 
 void pm_state_exit_post_ops(enum pm_state state, uint8_t substate_id)
 {
+#if !defined(CONFIG_SOC_NRF54H20_CPURAD)
+	if (state == PM_STATE_SUSPEND_TO_IDLE) {
+		soc_lrcconf_poweron_release(&pm_node, NRF_LRCCONF_POWER_DOMAIN_0);
+	}
+#endif
```

Comment:
This seems unneeded now. When exiting suspend to idle, the ACT PD should remain active.

Reply:
Couldn't we just simplify this and turn on the ACT PD on startup and just turn it off when entering s2ram or softoff? No need for idle hooks etc? You are right the active power domain needs to be up anyways except for entering suspend to ram or soft off